elide-contrib/elide-test-helpers/src/main/java/com/yahoo/elide/contrib/testhelpers/graphql/GraphQLDSL.java

```diff
@@ -0,0 +1,535 @@
+     */
+    public static Query query(String name, VariableDefinitions variableDefinitions, SelectionSet selectionSet) {
+        return new Query(name, variableDefinitions, selectionSet);
+    }
+
+    /**
+     * Constructs a named query
+     *
+     * @param selectionSet  Selected fields
+     *
+     * @return a named query
+     */
+    public static Query query(SelectionSet selectionSet) {
+        return new Query(null, null, selectionSet);
+    }
+
+    /**
+     * Creates a collection of GraphQL variable definitions.
+     *
+     * @param variableDefinitions  A variable-sized array of {@link VariableDefinition}s
+     *
+     * @return A collection of GraphQL variable definitions
+     *
+     * @see <a href="https://graphql.org/learn/queries/#variables">Variables</a>
+     */
+    public static VariableDefinitions variableDefinitions(VariableDefinition... variableDefinitions) {
+        return new VariableDefinitions(Arrays.asList(variableDefinitions));
+    }
+
+    /**
+     * Creates a single GraphQL variable definition without default value.
+     *
+     * @param variable  A string (without "$" sign) representing the name of the variable
+     * @param type  A simplified String representation of the aforementioned type literal.
+     *
+     * @return A single GraphQL variable definition
+     *
+     * @see <a href="https://graphql.org/learn/queries/#variables">Variables</a>
+     */
+    public static VariableDefinition variableDefinition(String variable, String type) {
+        return new VariableDefinition(variable, type, null);
+    }
+
+    /**
+     * Constructs a scalar response field.
+     *
+     * @param name  Field name
+     * @param value  Field value
+     *
+     * @return a field
+     */
+    public static Selection field(String name, String value) {
+        return new Field(name, Arguments.emptyArgument(), Field.quoteValue(value));
+    }
+
+    /**
+     * Constructs a scalar response field.
+     *
+     * @param name  Field name
+     * @param value  Field value
+     * @param quoted  Weather value is quoted
+     *
+     * @return a field
+     */
+    public static Selection field(String name, String value, boolean quoted) {
+        return new Field(name, Arguments.emptyArgument(), quoted ? Field.quoteValue(value) : value);
+    }
+
+    /**
+     * Creates a entity(object field) selection without {@link Argument}s.
+     *
+     * @param name  The name of the selected entity/field that would appear in a GraphQL query
+     * @param selectionSet  The fields of the entity that are selected
+     *
+     * @return a top-level selection
+     *
+     * @see <a href="https://graphql.org/learn/queries/#fields">Fields</a>
+     * @see <a href="https://graphql.org/learn/schema/#object-types-and-fields">Object Types and Fields</a>
+     */
+    public static Selection field(String name, Object... selectionSet) {
+        if (selectionSet.length == 1 && !(selectionSet[0] instanceof SelectionSet)) {
```

Comment:
I"m not following this.  A couple points:
 1. instanceof is a generally a code smell in Java (unless the code is doing reflection like the EntityDictionary).
 2.  It looks like this code is trying to infer from arguments if this is a query or a response.  The only time that should happen is inside the toQuery or toResponse functions (not during DSL construction).

Construction should be dumb/simple logic and serialization can be smarter.

Reply:
So it sounds like this is mixing two use cases:
1) A list of selection sets (but those aren't objects, they are selection sets>
2) A field where the value is a singular object.

These ought to be two different functions.